llvm/lib/Transforms/IPO/SCCP.cpp

```diff
@@ -357,8 +357,11 @@ static bool runIPSCCP(
     LLVM_DEBUG(dbgs() << "Found that GV '" << GV->getName()
                       << "' is constant!\n");
     while (!GV->use_empty()) {
-      StoreInst *SI = cast<StoreInst>(GV->user_back());
-      SI->eraseFromParent();
+      auto User = GV->user_back();
```

Comment:
I'd write this as `for (User *U : make_early_inc_range(GV->users()))`.

Reply:
thanks for advise!